pkg/util/net/dns/resolveconf_test.go

```diff
@@ -91,4 +91,42 @@ var _ = Describe("Resolveconf", func() {
 			Expect(domain).To(Equal("14wg5xngig6vzfqjww4kocnky3c9dqjpwkewzlwpf.com"))
 		})
 	})
+
+	Context("subdomain", func() {
+		It("should be added to the search domain entries", func() {
+			searchDomains := []string{"default.svc.cluster.local", "svc.cluster.local", "cluster.local"}
+
+			subdomain := "subdomain"
+			subdomainSearchDomain := SubdomainSearchDomain(searchDomains, subdomain)
+			if subdomainSearchDomain != "" {
+				searchDomains = append(searchDomains, subdomainSearchDomain)
+			}
+
+			Expect(searchDomains).To(Equal([]string{"default.svc.cluster.local",
+				"svc.cluster.local", "cluster.local", subdomain + ".default.svc.cluster.local"}))
+		})
```

Comment:
It may be clearer to the reader if you do something like this:
```
Expect(searchDomains).To(Equal(append(origSearchDomains, subdomain + "." + origSearchDomains[0]}))
```

Reply:
since the function is not appending anymore but just calculating the domain, 
and this is a unit test for the `DomainNameWithSubdomain` function, i am reducing it to

```
		It("should be added to the search domain longest entry", func() {
			searchDomains := []string{"default.svc.cluster.local", "svc.cluster.local", "cluster.local"}

			const subdomain = "subdomain"
			domain := DomainNameWithSubdomain(searchDomains, subdomain)
			Expect(domain).To(Equal(subdomain + "." + searchDomains[0]))
		})
```

Reply:
since the function is not appending anymore but just calculating the domain, 
and this is a unit test for the `DomainNameWithSubdomain` function, i am reducing it to

```
		It("should be added to the search domain longest entry", func() {
			searchDomains := []string{"default.svc.cluster.local", "svc.cluster.local", "cluster.local"}

			const subdomain = "subdomain"
			domain := DomainNameWithSubdomain(searchDomains, subdomain)
			Expect(domain).To(Equal(subdomain + "." + searchDomains[0]))
		})
```